hypothesis-python/src/hypothesis/searchstrategy/numbers.py

```diff
@@ -108,7 +108,9 @@ def __init__(self, allow_infinity, allow_nan):
         self.sampler = d.Sampler(weights)
 
     def __repr__(self):
-        return "%s()" % (self.__class__.__name__,)
+        return "FloatStrategy(allow_infinity={}, allow_nan={})".format(
```

Comment:
nit: Why have you hard-coded the class name instead of continuing to use `self.__class__.__name__`?